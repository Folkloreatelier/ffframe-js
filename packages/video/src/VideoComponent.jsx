import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import styles from './styles.scss';

const propTypes = {
    className: PropTypes.string,
};

const defaultProps = {
    className: null,
};

class VideoComponent extends Component {
    constructor(props) {
        super(props);

        this.onClickButton = this.onClickButton.bind(this);
    }

    onClickButton() {
        console.log('clicked');
    }

    render() {
        const { className } = this.props;
        return (
            <div
                className={classNames({
                    [styles.container]: true,
                    [className]: className !== null,
                })}
            >
                <button type="button" onClick={this.onClickButton}>
                    Button
                </button>
            </div>
        );
    }
}

VideoComponent.propTypes = propTypes;
VideoComponent.defaultProps = defaultProps;

export default VideoComponent;
