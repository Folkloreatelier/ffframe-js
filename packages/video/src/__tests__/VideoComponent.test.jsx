import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon';
import VideoComponent from '../VideoComponent';

test('match snapshot', () => {
    const component = renderer.create((
        <VideoComponent />
    ));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
