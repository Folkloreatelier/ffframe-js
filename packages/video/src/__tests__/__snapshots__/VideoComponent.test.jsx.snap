// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="undefined"
>
  <button
    onClick={[Function]}
    type="button"
  >
    Button
  </button>
</div>
`;
